keps/sig-node/3721-support-for-env-files/README.md

```diff
@@ -0,0 +1,546 @@
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [x] (R) Production readiness review completed
+- [x] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This is a proposal to add capability to populate environment variables in the
+container from a file.
+
+## Motivation
+
+Kubernetes provides existing mechanisms for setting environment variables within
+containers using ConfigMap and Secret resources. However, these mechanisms have
+a few drawbacks in following use-cases:
+
+1. An `initContainer` creates some configuration that needs to be consumed as an
+   environment variable by the main container. Existing mechanisms necessitate
+   additional ConfigMap or Secret API calls.
+
+2. The user is using a container offered by a vendor that requires configurating
+   environment variables (for eg, license key, one-time secret tokens). They can
+   use an `initContainer` to fetch this info into a file and the main container
+   can consume.
+
+The syntactic enhancement proposed in this KEP offers a simplified method for
+populating environment variables directly from files. This change would
+streamline the process for users.
+
+### Goals
+
+1. Support instantiating a container's environment variables from a file. This
+file must be in emptyDir volume. The env file can be created by an initContainer
+or sidecar in the emptyDir volume. kubelet will instantiate the env vars in the
+container from the specified file in the emptyDir volume but it will not mount
+the file.
```

Comment:
please make it clear that the volume will be mounted to the producer, but not consumer